drivers/sensor/icm20948/icm20948.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2023, Lazaro O'Farrill
+ * email: lazaroofarrill@gmail.com
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_ICM20948_H_
+#define ZEPHYR_DRIVERS_SENSOR_ICM20948_H_
+
+#include "zephyr/drivers/i2c.h"
+#include <stdint.h>
+
+struct icm20948_data {
+	int16_t accel_x;
+	int16_t accel_y;
+	int16_t accel_z;
+
+	int16_t temp;
+
+	int16_t gyro_x;
+	int16_t gyro_y;
+	int16_t gyro_z;
+
+	int16_t mag_x;
+	int16_t mag_y;
+	int16_t mag_z;
+	int16_t magn_scale_x;
+	int16_t magn_scale_y;
+	int16_t magn_scale_z;
+};
+
+struct icm20948_config {
+	struct i2c_dt_spec i2c;
+	uint8_t accel_fs;
+	uint8_t accel_hz;
+	uint8_t accel_fchoice;
+	uint8_t accel_lpf;
+	uint8_t gyro_fs;
+	uint8_t gyro_hz;
+	uint8_t gyro_fchoice;
+	uint8_t gyro_lpf;
+	uint8_t mag_freq;
+};
+
+typedef enum {
+	ACCEL_FS_SEL_2G = 0,
+	ACCEL_FS_SEL_4G = 1,
+	ACCEL_FS_SEL_8G = 2,
+	ACCEL_FS_SEL_16G = 3
+} accel_fs_sel;
+
+typedef enum {
+	GYRO_FS_250 = 0,
+	GYRO_FS_500 = 1,
+	GYRO_FS_1000 = 2,
+	GYRO_FS_2000 = 3,
+} gyro_fs_sel;
+
+typedef enum {
```

Comment:
typedefs are not really required, or :-) ?

Reply:
done